### PR TITLE
Bugfix/228 some errors not populting

### PR DESCRIPTION
**Pull request by yruslan**

### CODE REVIEW

File: conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala [hunk 1]
```diff
@@ -214,43 +213,20 @@ object DynamicInterpreter {
...
   }
 
+  /**
+    * Adds an explosion pseudo-rule before each group of more than 1 mapping rule.
+    * Adds a collection pseudo-rule after each group of more than 1 mapping rule
+    *
+    * @param ruleGroups A list of rules grouped together if they are mapping rules operating at the same array level
+    * @param schema     A schema of a dataset
+    * @return The new list of rules that might contain pseudo-rules
+    */
```
Comment (Zejnilovic) [inline conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala]: We talked that private API doesn't have to be commented, but I really appreciate you taking time and doing so.

Comment (yruslan) [reply conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala]: Thanks!

File: conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala [hunk 2]
```diff
@@ -214,43 +213,20 @@ object DynamicInterpreter {
...
-          pushGroup()
-          groups += List(a)
-      }
-    }
-    pushGroup()
-    groups.toList
+    Algorithms.stableGroupByOption[ConformanceRule, String](rules, {
+      case m: MappingConformanceRule => SchemaUtils.getDeepestArrayPath(schema, m.outputColumn)
+      case _ => None
+    }).map(_.toList).toList
```
Comment (GeorgiChochov) [inline conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala]: I get that you are mapping `Seq` to `List` here, but why not have the more abstract return type in the function: `Seq[Seq[ConformanceRule]]` we are not making use of some `List`-specific properties with the return of this function (we just `foldLeft` from what I could trace). I'd substitute it for the more abstract `Seq` type.

File: utils/src/main/scala/za/co/absa/enceladus/utils/general/Algorithms.scala [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    * A stable group by takes a collection of items and groups items when consecutive items have the same
+    * grouping value. A grouping value is provided by a callback function.
+    *
+    * The function does not change the original order of the elements.
+    *
+    * @param collection     An input collection
+    * @param groupedElement A callback that for a given item produces a value by which items should be grouped
+    * @return A sequence of groups of the original items
+    */
+  def stableGroupBy[T, A](collection: Seq[T], groupedElement: T => A): Seq[Seq[T]] = {
```
Comment (GeorgiChochov) [inline utils/src/main/scala/za/co/absa/enceladus/utils/general/Algorithms.scala]:
```suggestion
  def stableGroupBy[T, A](collection: Seq[T], groupBy: T => A): Seq[Seq[T]] = {
```
The second parameter is a callback function, I'd give it a more "verb-y" name  to reflect that. (The suggestion above is not the best option, I'm sure you can come up with something more appropriate)

File: conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala (hunk 2 above)
Comment (GeorgiChochov) [reply conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala]: Also, the implementation underneath is already a `List` when you look at the implementation of this algorithm.

File: utils/src/main/scala/za/co/absa/enceladus/utils/general/Algorithms.scala (hunk 3 above)
Comment (yruslan) [reply utils/src/main/scala/za/co/absa/enceladus/utils/general/Algorithms.scala]: In Scala standard library this callback function is named `f`. We can follow them...

**Pull request merged**